### PR TITLE
fix: duplicate envFrom in k8s manifests

### LINKED ISSUE #7740: Kubernetes extension generates environment variables twice
**Describe the bug**

I use the Kubernetes extension configuration properties to add environment variables to the `Deployment` / `DeploymentConfig`, and I'm getting the same piece of YAML generated twice.

This happens at least for generating env var from a ConfigMap and from a secret, both in the "one env var for specific key" and "env vars for all keys" scenarios.

For example, I have this config:

```properties
# one env var for specific key in the secret
quarkus.openshift.env-vars.MY_ENV_VAR.secret=the-secret
quarkus.openshift.env-vars.MY_ENV_VAR.value=key-in-the-secret

# env vars for all keys in the ConfigMap
quarkus.openshift.env-vars.IGNORED.configmap=the-configmap
```

With this, the generated YAML contains:

```yaml
    spec:
      containers:
      - env:
        - name: "MY_ENV_VAR"
          valueFrom:
            secretKeyRef:
              key: "key-in-the-secret"
              name: "the-secret"
              optional: false
        - name: "MY_ENV_VAR"
          valueFrom:
            secretKeyRef:
              key: "key-in-the-secret"
              name: "the-secret"
              optional: false
        envFrom:
        - configMapRef:
            name: "the-configmap"
            optional: false
        - configMapRef:
            name: "the-configmap"
            optional: false
```

In the debugger, I see `AddEnvVarDecorator.andThenVisit` called twice during `session.close()`, but as far as I was able to observe, the `AddEnvVarDecorator` is only added _once_ in the Quarkus extension.

**Expected behavior**
Only one generated piece of YAML for each config key.

**Actual behavior**
Two identical pieces of YAML for each config key.

**To Reproduce**
1. Add the `quarkus-kubernetes` extension.
2. Run `mvn clean package -DskipTests`.
3. See `target/kubernetes/*.yml`.

**Configuration**
See above.

### PR DESCRIPTION
**Pull request by iocanel**
This pull request fixes the presence of duplicate envFrom entries appearing in the manifests.

We recently fixes duplicate env variables, but mised this case.

Fixes: #7740

### CODE REVIEW

**Comment by geoand** [comment]
If I recall correctly, there was an issue about this right?

File: integration-tests/kubernetes/standard/src/test/java/io/quarkus/it/kubernetes/KubernetesWithEnvFromSecretTest.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        assertThat(kubernetesList.get(0)).isInstanceOfSatisfying(Deployment.class, d -> {
+            assertThat(d.getMetadata()).satisfies(m -> {
+                assertThat(m.getName()).isEqualTo("env-from-secret");
+            });
+
+            assertThat(d.getSpec()).satisfies(deploymentSpec -> {
+                assertThat(deploymentSpec.getTemplate()).satisfies(t -> {
+                    assertThat(t.getSpec()).satisfies(podSpec -> {
+                        assertThat(podSpec.getContainers()).hasOnlyOneElementSatisfying(container -> {
+                            assertThat(container.getEnvFrom()).hasOnlyOneElementSatisfying(env -> {
```
Comment (geoand) [inline integration-tests/kubernetes/standard/src/test/java/io/quarkus/it/kubernetes/KubernetesWithEnvFromSecretTest.java]: Maybe we should also assert that `container.getEnv()` is null?

Comment (geoand) [reply integration-tests/kubernetes/standard/src/test/java/io/quarkus/it/kubernetes/KubernetesWithEnvFromSecretTest.java]: Never mind, doesn't apply

**Comment by geoand** [comment]
@iocanel I want to merge it, but I want to know if we have an issue for this since it would be nice to link to it if it does exist.

**Comment by iocanel** [comment]
> @iocanel I want to merge it, but I want to know if we have an issue for this since it would be nice to link to it if it does exist.

Linked with #7740

**Comment by geoand** [comment]
Excellent, thanks!

**Review by geoand — APPROVED** [review]

**Pull request merged**